commit 7a887d2a1f91b8a0856735ddc8bf6d38b06e2b5a
Author: Christophe JAILLET <christophe.jaillet@wanadoo.fr>
Date:   Mon Jun 11 19:30:35 2018 +0200

    serial: pxa: Fix an error handling path in 'serial_pxa_probe()'
    
    [ Upstream commit 95a0e656580fab3128c7bee5f660c50784f53651 ]
    
    If port.line is out of range, we still need to release some resources, or
    we will leak them.
    
    Fixes: afc7851fab83 ("serial: pxa: Fix out-of-bounds access through serial port index")
    Signed-off-by: Christophe JAILLET <christophe.jaillet@wanadoo.fr>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
    Signed-off-by: Sasha Levin <alexander.levin@microsoft.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/tty/serial/pxa.c b/drivers/tty/serial/pxa.c
index eda3c7710d6a..4932b674f7ef 100644
--- a/drivers/tty/serial/pxa.c
+++ b/drivers/tty/serial/pxa.c
@@ -887,7 +887,8 @@ static int serial_pxa_probe(struct platform_device *dev)
 		goto err_clk;
 	if (sport->port.line >= ARRAY_SIZE(serial_pxa_ports)) {
 		dev_err(&dev->dev, "serial%d out of range\n", sport->port.line);
-		return -EINVAL;
+		ret = -EINVAL;
+		goto err_clk;
 	}
 	snprintf(sport->name, PXA_NAME_LEN - 1, "UART%d", sport->port.line + 1);
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1165_7a887d2a1f91b8a0856735ddc8bf6d38b06e2b5a.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1165_7a887d2a1f91b8a0856735ddc8bf6d38b06e2b5a.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
